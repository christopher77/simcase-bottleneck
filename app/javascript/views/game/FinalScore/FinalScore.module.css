.componentContainer {
  display: flex;
  flex-direction: column;
  width: 100%;
  align-items: center;
  justify-content: space-evenly;
  height: calc(100% - 114px);
  animation: animateProfileGate 450ms ease 1;
}
.componentContainer h1 {
  font-weight: 600;
  font-size: 22px;
  line-height: 27px;
}
.componentContainer h2 {
  font-weight: 600;
  font-size: 13px;
  line-height: 16px;
}

.finalScoreSection,
.userInformationSection {
  background-color: #ffffff;
  box-shadow: 0px 4px 40px rgba(0, 0, 0, 0.2);
  border-radius: 8px;
  width: 75%;
  padding: 10px;
}

.finalScoreSection {
  display: flex;
  justify-content: space-around;
}

.scoreItem {
  padding: 10px;
  font-weight: 800;
  font-size: 29px;
  line-height: 35px;
  text-align: center;
}

.titlesScore {
  font-weight: 400;
  font-size: 12px;
  line-height: 15px;
  color: #1a1212;
  text-align: center;
}

.componentContainer p {
  font-weight: 600;
  font-size: 13px;
  line-height: 16px;
  margin: 0 59px 0;
  text-align: center;
}

.userInformationSection {
  text-align: center;
}

.titleName {
  font-weight: 600;
  font-size: 13px;
  line-height: 16px;
  text-align: center;
  color: #4f4f4f;
  text-transform: uppercase;
}

.userInformationSection input {
  background-color: #ffffff;
  width: 80%;
  padding: 12px 20px;
  border: none;
  border-bottom: 2px solid black;
  font-size: 18px;
  text-align: center;
}

.componentContainer button {
  background: #73c056;
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
  border-radius: 47px;
  color: black;
  width: 10em;
  height: 42px;
  text-transform: uppercase;
  font-size: 17px;
  font-weight: bold;
  border-color: transparent;
  line-height: 21px;
}

.iconsContainer {
  display: flex;
  justify-content: space-around;
  width: 100%;
  bottom: 0;
  position: absolute;
  height: 60px;
}

.replay,
.list {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 42px;
  height: 42px;
  background-color: #73c056;
  box-shadow: 2px 1px 12px rgba(115, 192, 86, 0.8);
  border-radius: 21px;
}

.saveButton,
.replay,
.list {
  transition: all 0.5s;
  border-color: transparent;
}

.list img,
.replay img {
  width: 24px;
}
.saveButton:active,
.replay:active,
.list:active {
  transform: scale(1.1);
  box-shadow: 0px 8px 4px rgba(0, 0, 0, 0.25);
}
.saveButton:disabled {
  color: darkgray;
  background: lightgray;
}

@keyframes animateProfileGate {
  0% {
    opacity: 0;
    transform: scale(1.1);
  }
  100% {
    opacity: 1;
    transform: scale(1);
  }
}

@keyframes slide-top {
  0% {
    transform: translateY(100px);
  }
  100% {
    transform: translateY(0px);
  }
}
