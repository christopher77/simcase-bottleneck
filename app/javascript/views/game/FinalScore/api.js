export async function submitScore(score) {
  return fetch("http://localhost:3000/api/scores", {
    method: "POST",
    body: JSON.stringify(score),
    headers: {
      "Content-Type": "application/json"
    }
  })
    .then(resp => resp.json())
    .then(json => json)
    .catch(err => {});
}
