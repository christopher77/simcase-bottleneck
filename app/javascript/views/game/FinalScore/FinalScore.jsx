import React from "react";
import Header from "../../../components/Header/HeaderContainer";
import { navigate } from "@reach/router";
import styles from "./FinalScore.module.css";
import replayIcon from "../../../assets/replay-arrow.svg";
import listIcon from "../../../assets/list.svg";
import CurrencyFormat from "react-currency-format";
import { submitScore } from "./api";

function FinalScore({
  totalScore,
  successRate,
  reset,
  resetScores,
  addLastGame = () => {}
}) {
  const [username, setUsername] = React.useState("");
  const addUserLastGame = lastgame => addLastGame(lastgame);
  const [isEnabled, setIsEnabled] = React.useState(false);

  async function handleSubmit(event) {
    event.preventDefault();
    const dataToSend = {
      score: {
        score: totalScore,
        success_rate: successRate,
        user_attributes: { name: username }
      }
    };
    const lastgame = await submitScore(dataToSend);
    addUserLastGame(lastgame);
    navigate(`/leaderboard/${lastgame.id}`);
  }

  function handleChange(event) {
    event.preventDefault();
    const {
      target: { value }
    } = event;
    setUsername(value);
    if (value.length >= 2) {
      setIsEnabled(true);
    } else {
      setIsEnabled(false);
    }
  }

  function handleRedirectLeaderBoard() {
    reset();
    resetScores();
    navigate("/leaderboard");
  }

  function handlePlayAgain() {
    reset();
    resetScores();
    navigate("/pre-game");
  }

  return (
    <>
      <Header rightHeader={5} />
      <div className={styles.componentContainer}>
        <h1>Final Score</h1>
        <section className={styles.finalScoreSection}>
          <div>
            <div className={styles.scoreItem}>
              <CurrencyFormat
                value={totalScore}
                thousandSeparator={true}
                displayType={"text"}
              />
            </div>
            <div className={styles.titlesScore}>Pounds recycled</div>
          </div>
          <div>
            <div className={styles.scoreItem}>{successRate}</div>
            <div className={styles.titlesScore}>% of Max</div>
          </div>
        </section>

        <p>Save your score and see your position in the leaderboard</p>

        <section className={styles.userInformationSection}>
          <form onSubmit={handleSubmit}>
            <label className={styles.titleName} htmlFor="username">
              Enter your name:
            </label>
            <input
              type="text"
              autoComplete="off"
              name="username"
              id="username"
              aria-label="Enter your name"
              onChange={handleChange}
              value={username}
              required
              autoFocus={true}
            />
          </form>
        </section>
        <button
          className={styles.saveButton}
          onClick={handleSubmit}
          disabled={!isEnabled}
        >
          save my score
        </button>
      </div>
      <div className={styles.iconsContainer}>
        <button className={styles.replay} onClick={handlePlayAgain}>
          <img src={replayIcon} alt="replay-icon" />
        </button>
        <button className={styles.list} onClick={handleRedirectLeaderBoard}>
          <img src={listIcon} alt="list-icon" />
        </button>
      </div>
    </>
  );
}

export default FinalScore;
