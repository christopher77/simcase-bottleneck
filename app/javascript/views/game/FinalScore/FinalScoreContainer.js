import { connect } from "react-redux";
import FinalScore from "./FinalScore";
import { addLastGame } from "../../../redux/actions/UserActions";
import { resetScores } from "../../../redux/actions/UserActions";
import { reset } from "../../../redux/actions/RoundsActions";

const mapStateToProps = state => ({
  totalScore: state.user.totalScore,
  success: state.user.success,
  successRate: state.user.successRate
});
const mapDispatchToProps = {
  addLastGame: addLastGame,
  reset: reset,
  resetScores: resetScores
};
export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FinalScore);

// {
//   "totalScore": 0,
//   "success": 0,
//   "successRate": 0,
// }
