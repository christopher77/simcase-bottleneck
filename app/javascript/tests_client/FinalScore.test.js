import React from "react";
import { render, fireEvent } from "@testing-library/react";
import FinalScore from "../views/game/FinalScore/FinalScore";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";

const middlewares = [];
const mockStore = configureStore(middlewares);

const initialState = {
  user: {
    totalScore: 10,
    success: 10,
    successRate: 10
  },
  feedback: {
    feeback: false
  }
};

const store = mockStore(initialState);

function renderWithProvider(children) {
  return render(<Provider store={store} children={children} />);
}

describe("<FinalScore/>", () => {
  it("FinalScore component snapshot", () => {
    const { asFragment } = renderWithProvider(<FinalScore />);
    expect(asFragment).toMatchSnapshot();
  });

  it("submits", () => {
    fetch.mockResponseOnce(JSON.stringify({ id: 1 }));
    const { getByLabelText, getByText, debug } = renderWithProvider(
      <FinalScore />
    );
    const nameInput = getByLabelText("Enter your name:");
    fireEvent.change(nameInput, { target: { value: "yeee" } });
    const submitButton = getByText("save my score");
    fireEvent.click(submitButton);
    debug();
  });
});
