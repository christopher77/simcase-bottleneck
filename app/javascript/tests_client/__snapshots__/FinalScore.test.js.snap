// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FinalScore/> FinalScore component snapshot 1`] = `[Function]`;
